File: core/src/com/mygdx/game/Entity/PlayerEntity.java
package com.mygdx.game.Entity;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Animation;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;


/**
 * Created by Adrian on 2017-04-08.
 */

public class PlayerEntity extends MovableEntity {
    private static final int FRAME_COLS = 20;
    private Animation<TextureRegion> walkAnimation;
    private Texture walkSheet;
    private float stateTime;


    public PlayerEntity(String image,float x, float y) {
        super(image, x, y);
        walkSheet = new Texture(Gdx.files.internal("playerwalk.png"));
        TextureRegion[][] tmp = TextureRegion.split(walkSheet, (walkSheet.getWidth() / FRAME_COLS), walkSheet.getHeight());
        TextureRegion[] walkFrames = new TextureRegion[FRAME_COLS];

        int index = 0;
        for (int i = 0; i < FRAME_COLS; i++) {
            walkFrames[index] = tmp[0][i];
             index++;
        }
        walkAnimation = new Animation<>(0.050f, walkFrames);
        stateTime = 0f;
    }

     public void draw(SpriteBatch batch, float delta) {
        stateTime += delta;
        sprite.setPosition(position.x, position.y);
        TextureRegion currentFrame = walkAnimation.getKeyFrame(stateTime, true);
        batch.draw(currentFrame, position.x, position.y, sprite.getOriginX()+20, sprite.getOriginY()-20,
                (walkSheet.getWidth() / FRAME_COLS)/4, walkSheet.getHeight()/4, 1, 1, sprite.getRotation()+90);
    }

    @Override
    public void dispose() {
        super.dispose();
    }

    public void handleInput(float delta) {
        float ROTATION_SPEED = 6;

        if(Gdx.input.isKeyPressed(Input.Keys.LEFT) || Gdx.input.isKeyPressed(Input.Keys.A)) {
            sprite.setRotation(sprite.getRotation() + ROTATION_SPEED);
        } else if(Gdx.input.isKeyPressed(Input.Keys.RIGHT) || Gdx.input.isKeyPressed(Input.Keys.D)) {
            sprite.setRotation(sprite.getRotation() - ROTATION_SPEED);
        }

        if(Gdx.input.isKeyPressed(Input.Keys.UP) || Gdx.input.isKeyPressed(Input.Keys.W)) {
            forward(delta);
        } else if(Gdx.input.isKeyPressed(Input.Keys.DOWN) || Gdx.input.isKeyPressed(Input.Keys.A)) {
            backward(-delta);
        }
    }
}

File: core/src/com/mygdx/game/Entity/PlayerEntity.kt
package com.mygdx.game.Entity

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Input
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Animation
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.g2d.TextureRegion


/**
 * Created by Adrian on 2017-04-08.
 */

class PlayerEntity(image: String, x: Float, y: Float) : MovableEntity(image, x, y) {
    private val walkAnimation: Animation<TextureRegion>
    private val walkSheet: Texture
    private var stateTime: Float = 0.toFloat()


    init {
        walkSheet = Texture(Gdx.files.internal("playerwalk.png"))
        val tmp = TextureRegion.split(walkSheet, walkSheet.width / FRAME_COLS, walkSheet.height)
        val walkFrames = arrayOfNulls<TextureRegion>(FRAME_COLS)

        var index = 0
        for (i in 0 until FRAME_COLS) {
            walkFrames[index] = tmp[0][i]
            index++
        }
        walkAnimation = Animation<TextureRegion>(0.050f, *walkFrames)
        stateTime = 0f
    }

    fun draw(batch: SpriteBatch, delta: Float) {
        stateTime += delta
        sprite.setPosition(position.x, position.y)
        val currentFrame = walkAnimation.getKeyFrame(stateTime, true)
        batch.draw(currentFrame, position.x, position.y, sprite.originX + 20, sprite.originY - 20,
                (walkSheet.width / FRAME_COLS / 4).toFloat(), (walkSheet.height / 4).toFloat(), 1f, 1f, sprite.rotation + 90)
    }

    override fun dispose() {
        super.dispose()
    }

    fun handleInput(delta: Float) {
        val ROTATION_SPEED = 6f

        if (Gdx.input.isKeyPressed(Input.Keys.LEFT) || Gdx.input.isKeyPressed(Input.Keys.A)) {
            sprite.rotation = sprite.rotation + ROTATION_SPEED
        } else if (Gdx.input.isKeyPressed(Input.Keys.RIGHT) || Gdx.input.isKeyPressed(Input.Keys.D)) {
            sprite.rotation = sprite.rotation - ROTATION_SPEED
        }

        if (Gdx.input.isKeyPressed(Input.Keys.UP) || Gdx.input.isKeyPressed(Input.Keys.W)) {
            forward(delta)
        } else if (Gdx.input.isKeyPressed(Input.Keys.DOWN) || Gdx.input.isKeyPressed(Input.Keys.A)) {
            backward(-delta)
        }
    }

    companion object {
        private val FRAME_COLS = 20
    }
}